src/features/UsersAPI/UsersAPI.ts

```diff
@@ -0,0 +1,44 @@
+/* eslint-disable no-param-reassign */
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { User } from '../../types/User';
+import { getUsers } from '../../api/users';
+
+type UsersState = {
+  users: User[];
+  loading: boolean;
+  error: string;
+};
+
+const initialState: UsersState = {
+  users: [],
+  loading: false,
+  error: '',
+};
+
+export const loadUsers = createAsyncThunk('users/fetch', () => {
+  return getUsers();
+});
+
+const usersSlice = createSlice({
+  name: 'users',
+  initialState,
+  reducers: {},
+  extraReducers: builder => {
+    builder.addCase(loadUsers.pending, state => {
+      state.loading = true;
+    });
+
+    builder.addCase(loadUsers.fulfilled, (state, action) => {
+      state.users = action.payload;
+      state.loading = false;
+    });
+
+    builder.addCase(loadUsers.rejected, state => {
+      state.loading = false;
+      state.error = 'Error';
```

Comment:
In the 'loadUsers.rejected' case, it would be more informative to use the error message from the action payload instead of a static 'Error' string. Consider updating the error assignment to use 'action.error.message' or a similar property to provide more context about the failure.